### ISSUE TITLE
eslint --init: config.extends.push

### ISSUE THREAD

**Issue by GongT** [eslint/eslint#12707]
**Tell us about your environment**
Node version: v12.14.0
npm version: v6.13.4
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I want to enable eslint for my exists TypeScript project.

```bash
eslint --init
```
```text
? How would you like to use ESLint? To check syntax, find problems, and enforce code style
? What type of modules does your project use? JavaScript modules (import/export)
? Which framework does your project use? React
? Does your project use TypeScript? Yes
? Where does your code run? Browser, Node
? How would you like to define a style for your project? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? ./src/**.{ts,tsx}
? What format do you want your config file to be in? JavaScript
```

**What actually happened? Please include the actual, raw output from ESLint.**
Several module not found：
* @typescript-eslint/parser
* eslint-plugin-react
* @typescript-eslint

After install them all, It show me an error:
```text
Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .
Determining Config: 100% [==============================] 0.1s elapsed, eta 0.0s

Enabled 275 out of 275 rules based on 2 files.

config.extends.push is not a function
TypeError: config.extends.push is not a function
    at processAnswers (/MY_PROJ/node_modules/eslint/lib/init/config-initializer.js:331:24)
    at /MY_PROJ/node_modules/eslint/lib/init/config-initializer.js:615:28
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Are you willing to submit a pull request to fix this bug?**
No, I found the problem, but don't know how to fix it (in your way).

### selected "react":
> https://github.com/eslint/eslint/blob/master/lib/init/config-initializer.js#L294

### select "Inspect your JavaScript file(s)":    
> https://github.com/eslint/eslint/blob/master/lib/init/config-initializer.js#L326    
https://github.com/eslint/eslint/blob/master/lib/init/autoconfig.js#L336

`extendFromRecommended` will set `config.extends` to `RECOMMENDED_CONFIG_NAME`, there are two problems:
* RECOMMENDED_CONFIG_NAME is a string, after `extendFromRecommended` return, the parent scope expect `config.extends` is an array
* It drop my selection of "react" (or "vue")

### selected "typescript":
> https://github.com/eslint/eslint/blob/master/lib/init/config-initializer.js#L329

**Comment by trydalch** [comment]
I've run into the same problem trying to build the Typescript Node Starter project from Microsoft. There's no eslint config file, so I went to init it, and got the same error as @GongT

**Comment by yeonjuan** [comment]
I reproduced it- https://github.com/yeonjuan/autoconfbug.

**Comment by kaicataldo** [comment]
I get a totally different error when following the same steps. Seems like this is unfortunately quite broken at the moment.

```sh
Failed to load plugin 'react' declared in 'BaseConfig': Cannot find module 'eslint-plugin-react'
```

PRs to fix this are very welcome!

**Comment by koenpunt** [comment]
> I get a totally different error when following the same steps. Seems like this is unfortunately quite broken at the moment.

To fix that you should just add `eslint-plugin-react` to your dev dependencies;

    yarn add --dev eslint-plugin-react

Or

    npm i --save-dev eslint-plugin-react

**Comment by kaicataldo** [comment]
@koenpunt Check out the issue above. This is referring to the `eslint --init` functionality which should install deps for you.

**Comment by koenpunt** [comment]
> which should install deps for you.

Ah, didn't know this.

**Comment by mdjermanovic** [comment]
Reopening since there is PR #12738 to fix this.
